boot.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/deis/riak/chans"
+	"github.com/deis/riak/clustersrv"
+	"github.com/deis/riak/config"
+	"github.com/deis/riak/riak"
+)
+
+func main() {
+	conf, err := config.Get()
+	if err != nil {
+		log.Printf("Error getting config (%s)", err)
+		os.Exit(1)
+	}
+
+	cmdDoneCh := make(chan error)
+	serverDoneCh := make(chan error)
+	if !conf.RiakMaster {
+		// non-bootstrap nodes should start a riak server and join
+		log.Printf("Starting as a bootstrap node")
+		go func() {
+			httpClient := &http.Client{}
+			clusterServerURL := clustersrv.URLFromConfig(conf)
+			if err != nil {
+				cmdDoneCh <- err
+				return
+			}
+
+			if err := riak.Start(); err != nil {
+				cmdDoneCh <- err
+				return
+			}
+
+			if err := riak.Join(httpClient, clusterServerURL); err != nil {
+				cmdDoneCh <- err
+				return
+			}
+		}()
+	} else {
+		// bootstrap nodes should start (not join) a riak server and start the cluster server
+		log.Printf("Starting as a bootstrap node")
+
+		go func() {
+			if err := riak.Start(); err != nil {
+				cmdDoneCh <- err
+				return
+			}
+		}()
+		log.Printf("Cluster server starting on port %d", conf.ClusterServerHTTPPort)
+		go clustersrv.Start(conf.ClusterServerHTTPPort, serverDoneCh)
+	}
+
+	if err := chans.JoinErrs(serverDoneCh, cmdDoneCh); err != nil {
```

Comment:
If I read this right, if either the lock server or the riak server fails, the node will go down. What happens when the pod is brought back up? It will be a regular node and there will be no lock server?